host/config.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/containers/gvisor-tap-vsock/pkg/types"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	captureFile      = "capture.pcap"
+	localHost        = "127.0.0.1"
+	tapDeviceMacAddr = "5a:94:ef:e4:0c:ee"
+	gatewayMacAddr   = "5a:94:ef:e4:0c:dd"
+	defaultSubnet    = "192.168.127.0/24"
+	defaultMTU       = 1500
+)
+
+type arrayFlags []string
+
+func (i *arrayFlags) String() string {
+	return "Array Flags"
+}
+
+func (i *arrayFlags) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
+type subnet struct {
+	IP              string
+	GatewayIP       string
+	StaticDHCPLease map[string]string
+	StaticDNSHost   string
+	SubnetCIDR      string
+}
+
+func validateSubnet(s string) (*subnet, error) {
+	ip, _, err := net.ParseCIDR(s)
+	if err != nil {
+		return nil, errors.Wrap(err, "validating subnet")
+	}
+	ipv4 := ip.To4()
+	return &subnet{
+		IP:              ip.String(),
+		GatewayIP:       gwtIP(ipv4),
+		StaticDHCPLease: staticDHCP(ipv4),
+		StaticDNSHost:   staticDNSHost(ipv4),
+		SubnetCIDR:      s,
+	}, nil
+}
+
+// Gateway is always x.x.x.1
+func gwtIP(ip net.IP) string {
+	ip[3] = 1
```

Comment:
So how would that break things 🤔?   I receive an IP as **string** into `validateSubnet` within the function I update the last bit of 4-byte formated IPv4 to deterministic values and store them into individual **string** variables. Nothing is leaked out of the `validateSubnet` and everything is stored as strings.

Reply:
`gwtIP()`, `staticDHCP()`, and `staticDNSHost()` all modify the same IPv4 address in-place (and happens to be okay because the keep clobbering the same byte). While it's not an issue now, it's likely to lead to errors down the road because this behaviour is very non-obvious, so if we end up using the ip for other purposes later we may miss this and write buggy code.